parm/snow/ioda_bufr_python_encoder.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import bufr
+from pyioda.ioda.Engines.Bufr import Encoder
+
+
+def mask_container(container, mask):
+    new_container = bufr.DataContainer()
+    for var_name in container.list():
+        print(f" ... variable name: {var_name} ...")
+        var = container.get(var_name)
+        paths = container.get_paths(var_name)
+        new_container.add(var_name, var[mask], paths)
+
+    return new_container
+
+def create_obs_group(input_path):
+    YAML_PATH = "./obs/bufr_sfcsno_mapping.yaml"
+    container = bufr.Parser(input_path, YAML_PATH).parse()
+
+    sogr = container.get('variables/groundState')
+    snod = container.get('variables/totalSnowDepth')
+    snod[(sogr <= 11.0) | (sogr == 15.0)] = 0.0
```

Comment:
Without having seen the data, I think there should be an if statement here - so that we only replace the snod when it was undefined / missing. 

i.e., if snod > 0. (or some, slighty higher threshold), and sogr indictaes no snow cover, I assume we would want to use the snod value, not 0.

Reply:
Above (with an if statement) is all consistent with how the PR description.

Reply:
Good suggestions. I will investigate this. 

@rmclaren Do you have any suggestions?

Reply:
Thanks again for the good findings. I modified to only replace the filled/missing snod.